Signal/src/ViewControllers/MediaGalleryViewController.swift

```diff
@@ -264,10 +264,7 @@ class MediaGallery: NSObject, MediaGalleryDataSource, MediaTileViewControllerDel
 
         self.options = options
         self.mediaGalleryFinder = OWSMediaGalleryFinder(thread: thread)
-        let navController = MediaGalleryNavigationController()
-        self.navigationController = navController
         super.init()
-        navController.retainUntilDismissed = self
```

Comment:
We can't safely allocate `MediaGalleryNavigationController` in the initializer, since we only hold a weak reference to it.  We need to allocate it right before present it.

Reply:
Yeah, you're right, that was dumb.